src/rules/createRegularRule.js

```diff
@@ -0,0 +1,145 @@
+import {clone, toCSS, findClassNames} from '../utils'
+import createBaseRule from './createBaseRule'
+
+/**
+ * Create a regular rule.
+ *
+ * @param {String} selector
+ * @param {Object} style
+ * @param {Object} options
+ * @return {Object}
+ */
+export default function createRegularRule(selector = '', style, options) {
+  const rule = createBaseRule('regular', {
+    className: '',
+    originalStyle: style,
+    // We expect style to be plain object.
+    style: clone(style),
+    options
+  })
+
+  if (options.named) {
+    rule.name = selector
+    rule.className = options.className || (rule.name ? `${rule.name}--${rule.id}` : rule.id)
+    selector = `.${rule.className}`
+  }
+
+  /**
+   * Get/set selector string.
+   *
+   * Attenition: use setter with caution. Most browser didn't implement selector
+   * text setter, so it will result in rerendering of entire style sheet.
+   *
+   * @return {String}
+   */
+  Object.defineProperty(rule, 'selector', {
+    get: () => {
+      if (rule.renderable) {
+        return rule.options.Renderer.getSelector(rule.renderable)
+      }
+
+      return selector
+    },
+
+    set: (newSelector = '') => {
```

Comment:
I believe the whole function will be deoptimized: https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#2-unsupported-syntax